### ISSUE TITLE
Weather Station Not working

### ISSUE THREAD

**Issue by niraviry** [home-assistant/home-assistant.github.io#18596]
### Feedback

Since few versions ago (might be 2021.3.4 or before) KNX weather does not work.
The weather variables were defined long time ago and stopped working since the above version.
The weather station is from Schneider Electric (Merten). It is not so popular but it works great.

I have enabled debugging for the KNX, and I see bus messages of the weather station for all of my sensors (light, temp, rain and wind).
I also get the night lights turn on and off automatically at the set illumination value inside the KNX bus.
It is clear that data is being delivered to the HA but no entities (sensors) are created anymore.
In the developers tools, I do not see the sensors exist. They vanished.
This looks like a bug.

### URL

https://www.home-assistant.io/integrations/knx/

### Version

2021.7.4

**Comment by frenck** [comment]
Hi there @niraviry 👋 

It seems like you want to report an issue with an Home Assistant integration, however, you have submitted documentation feedback instead. Please report issues with Home Assistant integrations here:

<https://github.com/home-assistant/core/issues>

Thanks 👍 

../Frenck

**Comment by farmio** [comment]
Hi! Before you create this issue elsewhere again be sure to read breaking changes of your updates https://www.home-assistant.io/blog/2021/06/02/release-20216/
TLDR: define sensors in yaml config. `create_sensors` is deprecated - as your log has probably told you.

**Comment by niraviry** [comment]
Hi Matthias.

I have read the breaking changes. The KNX Climate breaking change is
irrelevant. I am talking about my weather station (a Schneider KNX one) and
not climate devices. The 2021.6 did not kill the sensors. It happened a few
versions before 2021.6.
I define the KNX weather exactly as in the documentation.
I do not get any error from this definition. I have enabled the debug and
see messages as should be (outdoor light, wind speed, rain and temp). I do
not see any sensor in the entity list.
BTW, the majority of users have ABB KNX so it does not help as I suspect it
has something to do with that fact.

Thanks,

Nir Aviry

‫בתאריך שבת, 24 ביולי 2021 ב-20:51 מאת ‪Matthias Alphart‬‏ <‪
***@***.***‬‏>:‬

**Comment by farmio** [comment]
Read the whole section. 

> The create_sensors option for weather entities is now deprecated. Please use template sensors instead.
>This option created sensor entities out of weather configuration. This will not work anymore when sensor entities are also only generated from configuration.


https://github.com/home-assistant/core/pull/49640

**Comment by niraviry** [comment]
OK. Got it.

Thanks.

‫בתאריך יום א׳, 25 ביולי 2021 ב-21:35 מאת ‪Matthias Alphart‬‏ <‪
***@***.***‬‏>:‬

**Comment by niraviry** [comment]
H
Tried to figure out how to hook a sensor template to KNX and I am lost.
All KNX documentation does not show any change. What I have in my
configuration.yaml is what appears in the documentation as follows:
knx: weather: - name: "Home" address_temperature: "7/0/0"
address_brightness_south: "7/0/1" address_brightness_west: "7/0/2"
address_brightness_east: "7/0/3" address_brightness_north: "7/0/11"

I also did not find any examples. I can create a sensor using a template
but I do not see how to make it a KNX one.

Please help.


‫בתאריך יום א׳, 25 ביולי 2021 ב-21:35 מאת ‪Matthias Alphart‬‏ <‪
***@***.***‬‏>:‬

**Comment by farmio** [comment]
Just create normal knx sensors instead of templates. This information was somewhat misleading.

Please stop using this issue tracker as a support channel. There is a community forum discussing this https://community.home-assistant.io/t/knx-cookbook/230972/251?u=farmio or you can find help at the HA discord server.

**Comment by niraviry** [comment]
Thanks.

‫בתאריך יום ב׳, 26 ביולי 2021 ב-13:32 מאת ‪Matthias Alphart‬‏ <‪
***@***.***‬‏>:‬
